src/EditorFeatures/Core/AddUsings/IAutomaticallyAddMissingImportsService.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.AddMissingImports
+{
+    internal interface IAutomaticallyAddMissingImportsService : IWorkspaceService
+    {
+        void AddMissingImports(Document document, TextSpan textSpan, IUIThreadOperationContext operationContext);
```

Comment:
interesitng that this is a workspace service. would we expect different impls per workspace?

Reply:
Likely this can just go through MEF import/export directly instead of being a workspace service. The `IUIThreadOperationContext` is VS specific, and if implemented in another environment might be different. Not tied to workspace implementations.

Reply:
Actually, it would be best to just have all of this in EditorFeatures since VS4M will want it as well.